python/ray/autoscaler/_private/gcp/config.py

```diff
@@ -367,23 +378,18 @@ def _configure_subnet(config, compute):
     # work out-of-the-box
     default_subnet = subnets[0]
 
-    if "networkInterfaces" not in config["head_node"]:
-        config["head_node"]["networkInterfaces"] = [{
-            "subnetwork": default_subnet["selfLink"],
-            "accessConfigs": [{
-                "name": "External NAT",
-                "type": "ONE_TO_ONE_NAT",
-            }],
-        }]
-
-    if "networkInterfaces" not in config["worker_nodes"]:
-        config["worker_nodes"]["networkInterfaces"] = [{
-            "subnetwork": default_subnet["selfLink"],
-            "accessConfigs": [{
-                "name": "External NAT",
-                "type": "ONE_TO_ONE_NAT",
-            }],
-        }]
+    default_interfaces = [{
+        "subnetwork": default_subnet["selfLink"],
+        "accessConfigs": [{
+            "name": "External NAT",
+            "type": "ONE_TO_ONE_NAT",
+        }],
+    }]
+
+    for node_config in node_configs:
+        if "networkInterfaces" not in node_config:
```

Comment:
Can you add a test for this function to verify that the behavior is maintained before & after this change?

Reply:
I'll add a test that verifies this piece of data is inserted when not all networkInterfaces fields are specified and not inserted when they are specified.

Reply:
Hmm, I meant "check that missing networkInterface fields are filled and list subnets isn't called if everything is already configured"